src/presentation_exchange.rs

```diff
@@ -1,80 +1,846 @@
 use serde_json::Map;
+use ssi_claims::{
+    jwt::{AnyRegisteredClaim, Issuer, RegisteredClaim, VerifiablePresentation},
+    CompactJWSString, VerificationParameters,
+};
+use ssi_dids::{
+    ssi_json_ld::{
+        object::value::FragmentRef,
+        syntax::{from_value, Value},
+    },
+    VerificationMethodDIDResolver, DIDJWK,
+};
+use ssi_jwk::JWK;
+use ssi_verification_methods::AnyJwkMethod;
+
+/// A JSONPath is a string that represents a path to a specific value within a JSON object.
+///
+/// For syntax details, see [https://identity.foundation/presentation-exchange/spec/v2.0.0/#jsonpath-syntax-definition](https://identity.foundation/presentation-exchange/spec/v2.0.0/#jsonpath-syntax-definition)
+pub type JsonPath = String;
+
+/// The predicate Feature introduces properties enabling Verifier to request that Holder apply a predicate and return the result.
+///
+/// The predicate Feature extends the Input Descriptor Object `constraints.fields` object to add a predicate property.
+///
+/// The value of predicate **MUST** be one of the following strings: `required` or `preferred`.
+///
+/// If the predicate property is not present, a Conformant Consumer **MUST NOT** return derived predicate values.
+///
+/// See: [https://identity.foundation/presentation-exchange/#predicate-feature](https://identity.foundation/presentation-exchange/#predicate-feature)
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub enum Predicate {
+    /// required - This indicates that the returned value **MUST** be the boolean result of
+    /// applying the value of the filter property to the result of evaluating the path property.
+    #[serde(rename = "required")]
+    Required,
+    /// preferred - This indicates that the returned value **SHOULD** be the boolean result of
+    /// applying the value of the filter property to the result of evaluating the path property.
+    #[serde(rename = "preferred")]
+    Preferred,
+}
 
+/// The Presentation Definition MAY include a format property. The value MUST be an object with one or
+/// more properties matching the registered [ClaimFormatDesignation] (e.g., jwt, jwt_vc, jwt_vp, etc.).
+/// The properties inform the Holder of the Claim format configurations the Verifier can process.
+/// The value for each claim format property MUST be an object composed as follows:
+///
+/// The object MUST include a format-specific property (i.e., alg, proof_type) that expresses which
+/// algorithms the Verifier supports for the format. Its value MUST be an array of one or more
+/// format-specific algorithmic identifier references, as noted in the [ClaimFormatDesignation].
+///
+/// See [https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-definition](https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-definition)
+/// for an example schema.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum ClaimFormat {
```

Comment:
There is a bug when serializing and deserializing the `ClaimFormat` type, resulting in `JwtVcJson` being deserialized into `JwtVpJson`.

This occurs in the e2e test. Re-arranging the enum field order does change the behavior, but does not fix the bug.

Marking this problem as it will have significant side effects and is difficult to detect. So it should be tracked for a bugfix.

Reply:
This has been resolved. Removing the `untagged` seems to do the trick.